Pod linting fails on XCode 14 with watchOS platform

* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

Ran `pod lib lint` on my [pod](https://github.com/SwiftyLab/AsyncObjects/tree/obj-wait-err) with Xcode 14.

```shell
pod lib lint --platforms=watchos --no-clean --allow-warnings --verbose
```

## What did you expect to happen?

I expected CocoaPods to run linting and tests and succeed if tests passed.

## What happened instead?

Although tests succeeded, the command indicated failure with following results:

```
    ** TEST SUCCEEDED **
    
    Testing started
 -> AsyncObjects (1.0.0)
    - WARN  | keys: Unrecognized `parallelizable` key for `scheme` attribute.
    - ERROR | [watchOS] xcodebuild: Returned an unsuccessful exit code.
    - NOTE  | [watchOS] xcodebuild:  xcodebuild: error: Unable to find a destination matching the provided destination specifier:
    - NOTE  | xcodebuild:  note: Using codesigning identity override: -
    - NOTE  | [watchOS] xcodebuild:  note: Building targets in dependency order
    - NOTE  | [watchOS] xcodebuild:  Pods.xcodeproj: warning: The watchOS Simulator deployment target 'WATCHOS_DEPLOYMENT_TARGET' is set to 2.0, but the range of supported deployment target versions is 4.0 to 9.0.99. (in target 'OrderedCollections' from project 'Pods')
    - NOTE  | [watchOS] xcodebuild:  note: Metadata extraction skipped. No AppIntents.framework dependency found. (in target 'OrderedCollections' from project 'Pods')
    - WARN  | [watchOS] xcodebuild:  /Users/soumyaranjanmahunt/Documents/personal_projs/AsyncObjects/Tests/AsyncObjectsTests/StandardLibraryTests.swift:33:19: warning: immutable value 'val' was never used; consider replacing with '_' or removing it
    - NOTE  | [watchOS] xcodebuild:  note: Metadata extraction skipped. No AppIntents.framework dependency found. (in target 'AsyncObjects' from project 'Pods')
    - WARN  | [watchOS] xcodebuild:  /Users/soumyaranjanmahunt/Documents/personal_projs/AsyncObjects/Sources/AsyncObjects/CancellationSource.swift:6:62: warning: Topic reference 'init(linkedWith%3A)' couldn't be resolved. No local documentation matches this reference.
    - NOTE  | [watchOS] xcodebuild:  note: Metadata extraction skipped. No AppIntents.framework dependency found. (in target 'AsyncObjects-Unit-Tests' from project 'Pods')
```

On closer inspection I witnessed error is happening when building with `xcodebuild`:

```
-> Pod installation complete! There are 2 dependencies from the Podfile and 2 total pods installed.
   Building with `xcodebuild`. 
    $ /usr/bin/xcodebuild clean build -workspace
    /var/folders/fr/vlmkwkt93s9_0xv6q4hfmppr0000gp/T/CocoaPods-Lint-20220924-88954-1rh4pmz-AsyncObjects/App.xcworkspace
    -scheme App -configuration Release CODE_SIGN_IDENTITY=- -sdk watchsimulator
    -destination id=B01F93BB-ECD5-4F21-9BB9-6C3AFC153A51
    Command line invocation:
        /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild clean build -workspace /var/folders/fr/vlmkwkt93s9_0xv6q4hfmppr0000gp/T/CocoaPods-Lint-20220924-88954-1rh4pmz-AsyncObjects/App.xcworkspace -scheme App -configuration Release CODE_SIGN_IDENTITY=- -sdk watchsimulator -destination id=B01F93BB-ECD5-4F21-9BB9-6C3AFC153A51
    
    User defaults from command line:
        IDEPackageSupportUseBuiltinSCM = YES
    
    Build settings from command line:
        CODE_SIGN_IDENTITY = -
        SDKROOT = watchsimulator9.0
    
    2022-09-24 20:18:42.094 xcodebuild[88976:1465276] Writing error result bundle to /var/folders/fr/vlmkwkt93s9_0xv6q4hfmppr0000gp/T/ResultBundle_2022-24-09_20-18-0042.xcresult
    xcodebuild: error: Unable to find a destination matching the provided destination specifier:
    		{ id:B01F93BB-ECD5-4F21-9BB9-6C3AFC153A51 }
    
    	The requested device could not be found because no available devices matched the request.
    
    	Available destinations for the "App" scheme:
    		{ platform:watchOS, id:dvtdevice-DVTiOSDevicePlaceholder-watchos:placeholder, name:Any watchOS Device }
    		{ platform:watchOS Simulator, id:dvtdevice-DVTiOSDeviceSimulatorPlaceholder-watchsimulator:placeholder, name:Any watchOS Simulator Device }
    		{ platform:watchOS Simulator, id:B01F93BB-ECD5-4F21-9BB9-6C3AFC153A51, OS:9.0, name:Apple Watch SE (44mm) (2nd generation) }
    		{ platform:watchOS Simulator, id:A40285F8-8247-49C6-AAFD-321B88E41658, OS:9.0, name:Apple Watch Series 5 (40mm) }
    		{ platform:watchOS Simulator, id:AB1C2816-BDFA-4D50-B790-69FA84D167DB, OS:9.0, name:Apple Watch Series 5 (44mm) }
    		{ platform:watchOS Simulator, id:71CC520D-FD75-43D7-B4E1-08E12F5ED52F, OS:9.0, name:Apple Watch Series 6 (40mm) }
    		{ platform:watchOS Simulator, id:E08E9ACD-7C2F-4256-B562-B29EDA91D679, OS:9.0, name:Apple Watch Series 6 (44mm) }
    		{ platform:watchOS Simulator, id:03CD2A9B-050D-4A1A-AEFF-C51504B775A7, OS:9.0, name:Apple Watch Series 7 (41mm) }
    		{ platform:watchOS Simulator, id:AE2DBAA4-BC7E-4FAD-96EE-887B14B78777, OS:9.0, name:Apple Watch Series 7 (45mm) }
    		{ platform:watchOS Simulator, id:737D2762-B22C-4998-85DE-EAB420085D88, OS:9.0, name:Apple Watch Series 8 (41mm) }
    		{ platform:watchOS Simulator, id:DBC3DAD8-1301-4DC3-B7FA-40952ED9966A, OS:9.0, name:Apple Watch Series 8 (45mm) }
    		{ platform:watchOS Simulator, id:C02DCFDF-D999-46E1-B53C-2A4C34B0EFA5, OS:9.0, name:Apple Watch Ultra (49mm) }
    		{ platform:iOS Simulator, id:3F0AC861-4587-44D9-B50D-5AB9E7D40696, OS:16.0, name:iPhone 14 }
    		{ platform:iOS Simulator, id:0F5297C4-5415-4897-AAA9-053510458D4C, OS:16.0, name:iPhone 14 Plus }
    		{ platform:iOS Simulator, id:E3E51727-DF1B-4FB9-ACA6-98B03C170EC6, OS:16.0, name:iPhone 14 Pro }
    		{ platform:iOS Simulator, id:0E1252BA-119A-449E-96A8-3BBA90BE4A66, OS:16.0, name:iPhone 14 Pro Max }
```

This error doesn't make sense as the destination id that is said to be unavailable is present in available destination list for app scheme. And this error only happens for `watchOS` platform. The complete log when running with GitHub action. available [here](https://github.com/SwiftyLab/AsyncObjects/actions/runs/3118403707/jobs/5057669824#step:5:111).

## CocoaPods Environment

### Stack

```
   CocoaPods : 1.11.3
        Ruby : ruby 2.6.8p205 (2021-07-07 revision 67951) [universal.arm64e-darwin21]
    RubyGems : 3.0.3.1
        Host : macOS 12.6 (21G115)
       Xcode : 14.0 (14A309)
         Git : git version 2.37.0 (Apple Git-136)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib
Repositories : cocoapods - git - https://github.com/CocoaPods/Specs.git @ 38515a13a0e7b1cdfb3f38e19289ca754bd24b09

               trunk - CDN - https://cdn.cocoapods.org/
```

### Installation Source

```
Executable Path: /usr/local/bin/pod
```

### Plugins

```
claide-plugins                        : 0.9.2
cocoapods-deintegrate                 : 1.0.5
cocoapods-dependencies                : 1.3.0
cocoapods-disable-podfile-validations : 0.1.1
cocoapods-embed-flutter               : 0.6.1
cocoapods-generate                    : 2.2.2
cocoapods-packager                    : 1.5.0
cocoapods-plugins                     : 1.0.0
cocoapods-search                      : 1.0.1
cocoapods-stats                       : 1.1.0
cocoapods-trunk                       : 1.6.0
cocoapods-try                         : 1.2.0
cocoapods_debug                       : 0.1.0
```

## Project that demonstrates the issue

Demonstration project: https://github.com/SwiftyLab/AsyncObjects/tree/obj-wait-err, use the same branch as in the url.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -60,6 +60,10 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
   [Seth Friedman](https://github.com/sethfri)
   [#11235](https://github.com/CocoaPods/CocoaPods/pull/11235)
 
+* Fix publishing podspecs with watchOS support on Xcode 14
+  [Justin Martin](https://github.com/justinseanmartin)
+  [#11660](https://github.com/CocoaPods/CocoaPods/pull/11660)
+
 ## 1.11.3 (2022-03-16)
 
 ##### Enhancements

--- a/lib/cocoapods/validator.rb
+++ b/lib/cocoapods/validator.rb
@@ -1098,7 +1098,6 @@ module Pod
         end
       when :watchos
         command += %w(CODE_SIGN_IDENTITY=- -sdk watchsimulator)
-        command += Fourflusher::SimControl.new.destination(:oldest, 'watchOS', deployment_target)
       when :tvos
         command += %w(CODE_SIGN_IDENTITY=- -sdk appletvsimulator)
         command += Fourflusher::SimControl.new.destination(:oldest, 'tvOS', deployment_target)

--- a/spec/unit/validator_spec.rb
+++ b/spec/unit/validator_spec.rb
@@ -821,7 +821,7 @@ module Pod
         Executable.expects(:execute_command).with('xcodebuild', command + args, true).once.returns('')
         args = %w(CODE_SIGN_IDENTITY=- -sdk iphonesimulator) + Fourflusher::SimControl.new.destination('iPhone 4s')
         Executable.expects(:execute_command).with('xcodebuild', command + args, true).once.returns('')
-        args = %w(CODE_SIGN_IDENTITY=- -sdk watchsimulator) + Fourflusher::SimControl.new.destination('Apple Watch - 38mm')
+        args = %w(CODE_SIGN_IDENTITY=- -sdk watchsimulator)
         Executable.expects(:execute_command).with('xcodebuild', command + args, true).once.returns('')
         validator.validate.should == true
       end
@@ -850,7 +850,7 @@ module Pod
         Executable.expects(:execute_command).with('xcodebuild', command + args, true).once.returns('')
         args = %w(CODE_SIGN_IDENTITY=- -sdk iphonesimulator) + Fourflusher::SimControl.new.destination('iPhone 4s')
         Executable.expects(:execute_command).with('xcodebuild', command + args, true).once.returns('')
-        args = %w(CODE_SIGN_IDENTITY=- -sdk watchsimulator) + Fourflusher::SimControl.new.destination('Apple Watch - 38mm')
+        args = %w(CODE_SIGN_IDENTITY=- -sdk watchsimulator)
         Executable.expects(:execute_command).with('xcodebuild', command + args, true).once.returns('')
         validator.validate.should == true
       end
@@ -881,7 +881,7 @@ module Pod
         Executable.expects(:execute_command).with('xcodebuild', command + args, true).once.returns('')
         args = %w(CODE_SIGN_IDENTITY=- -sdk iphonesimulator) + Fourflusher::SimControl.new.destination('iPhone 4s') + analyzer_args
         Executable.expects(:execute_command).with('xcodebuild', command + args, true).once.returns('')
-        args = %w(CODE_SIGN_IDENTITY=- -sdk watchsimulator) + Fourflusher::SimControl.new.destination('Apple Watch - 38mm') + analyzer_args
+        args = %w(CODE_SIGN_IDENTITY=- -sdk watchsimulator) + analyzer_args
         Executable.expects(:execute_command).with('xcodebuild', command + args, true).once.returns('')
         validator.validate.should == true
       end
@@ -913,7 +913,7 @@ module Pod
         Executable.expects(:execute_command).with('xcodebuild', command + args, true).once.returns('')
         args = %w(CODE_SIGN_IDENTITY=- -sdk iphonesimulator) + Fourflusher::SimControl.new.destination('iPhone 4s') + analyzer_args
         Executable.expects(:execute_command).with('xcodebuild', command + args, true).once.returns('')
-        args = %w(CODE_SIGN_IDENTITY=- -sdk watchsimulator) + Fourflusher::SimControl.new.destination('Apple Watch - 38mm') + analyzer_args
+        args = %w(CODE_SIGN_IDENTITY=- -sdk watchsimulator) + analyzer_args
         Executable.expects(:execute_command).with('xcodebuild', command + args, true).once.returns('')
         validator.validate.should == true
       end
@@ -937,7 +937,7 @@ module Pod
         Executable.expects(:execute_command).with('xcodebuild', command + args, true).once.returns('')
         args = %w(CODE_SIGN_IDENTITY=- -sdk iphonesimulator) + Fourflusher::SimControl.new.destination('iPhone 4s')
         Executable.expects(:execute_command).with('xcodebuild', command + args, true).once.returns('')
-        args = %w(CODE_SIGN_IDENTITY=- -sdk watchsimulator) + Fourflusher::SimControl.new.destination('Apple Watch - 38mm')
+        args = %w(CODE_SIGN_IDENTITY=- -sdk watchsimulator)
         Executable.expects(:execute_command).with('xcodebuild', command + args, true).once.returns('')
         validator.validate
       end

